contracts/deploy/052_upgrade_buyback.js

```diff
@@ -0,0 +1,148 @@
+      recentBuybackAddress
+    );
+    const ousdBalance = await cOUSD.balanceOf(recentBuybackAddress);
+
+    // Deploy Buyback proxy
+    const dBuybackProxy = await deployWithConfirmation("BuybackProxy");
+    const cBuybackProxy = await ethers.getContractAt(
+      "BuybackProxy",
+      dBuybackProxy.address
+    );
+
+    // Deploy new Buyback implementation contract
+    const dBuybackImpl = await deployWithConfirmation("Buyback");
+
+    // Initialize proxy contract
+    await withConfirmation(
+      cBuybackProxy
+        .connect(sDeployer)
+        ["initialize(address,address,bytes)"](
+          dBuybackImpl.address,
+          deployerAddr,
+          [],
+          await getTxOpts()
+        )
+    );
+
+    const cBuyback = await ethers.getContractAt(
+      "Buyback",
+      cBuybackProxy.address
+    );
+
+    // Initialize implementation contract
+    const initFunction =
+      "initialize(address,address,address,address,address,address,address,address,uint256)";
+    await withConfirmation(
+      cBuyback.connect(sDeployer)[initFunction](
+        assetAddresses.uniswapV3Router,
+        strategistAddr,
+        strategistAddr, // Treasury manager
+        cOUSD.address,
+        assetAddresses.OGV,
+        assetAddresses.USDT,
+        assetAddresses.WETH,
+        assetAddresses.RewardsSource,
+        "5000", // 50%
+        await getTxOpts()
+      )
+    );
+
+    // Transfer governance of new contract to the governor
+    await withConfirmation(
+      cBuyback.connect(sDeployer).transferGovernance(timelockAddr)
+    );
+
+    // Governance Actions
+    // ----------------
+    return {
+      name: "Upgrade Buyback",
+      actions: [
+        {
+          // 1. Increase performance fee to 20%
+          contract: cVaultAdmin,
+          signature: "setTrusteeFeeBps(uint256)",
+          args: ["2000"], // 20%
+        },
+        {
+          // 2. Claim Governance of the new Buyback contract
+          contract: cBuyback,
+          signature: "claimGovernance()",
+        },
+        {
+          // 3. Update trustee address on Vault
+          contract: cVaultAdmin,
+          signature: "setTrusteeAddress(address)",
+          args: [cBuyback.address],
+        },
+        {
+          // 4. Transfer OUSD balance to Governor
+          contract: cRecentBuyback,
+          signature: "transferToken(address,uint256)",
+          args: [cOUSDProxy.address, ousdBalance],
+        },
+        {
+          // 5. Transfer OGN balance to Governor
+          contract: cOldBuyback,
+          signature: "transferToken(address,uint256)",
+          args: [cOGN.address, ognBalance],
+        },
+        {
+          // 6. Transfer OUSD balance from Governor to New Contract
+          contract: cOUSD,
+          signature: "transfer(address,uint256)",
+          args: [cBuyback.address, ousdBalance],
+        },
+        {
+          // 7. Transfer OGN balance from Governor to New Contract
```

Comment:
Since the community has voted on spending the OGN, let's have the old OGN end up on the strategist multisig, rather than the new buyback.